README.md

```diff
@@ -26,11 +26,17 @@ When the reader has completed this Code Pattern, they will understand how to:
 * Generate captions for an image using the MAX Model's REST API
 * Run a web application that using the model's REST API
 
-<!-- TODO: Insert Arch diagram -->
+![Architecture](doc/source/images/architecture.jpg)
 
 ## Flow
 
-<!-- TODO: Add Flow Steps -->
+1. Server initializes by sending default images to Model API and receiving caption data.
+2. Server starts Web UI with default images and caption data.
+3. User interacts with Web UI containing default images and caption data.
+4. User uploads image(s) to the Server via Web UI.
+5. Server sends image(s) to Model API and receives caption data.
+6. Server returns new caption data to Web UI.
+7. Web UI displays the new image(s) and caption data.
 
```

Comment:
What you have will work, but I think the diagram is too busy, and you can combine some things into a single step. Maybe have 4 steps, 1 & 4 between server and API, 2 between user and UI, 3 between UI and server. Then for flow, maybe something like:
1) Server requests and receives default images from API
2) User interacts with UI to view default images and request data on uploaded images
3) UI requests image data from server and displays results
4) Server gets caption data from Model API

Reply:
I'll try minimizing it a bit more based on this but there's a one incorrect statement in your suggestions though. The defaults images start on the server and it sends them to the model API. I'll take another pass at the flow based on this after I grab lunch.

Reply:
Sounds good. My flow steps were just to imply relationships. Counting on you to make the correct the content.

Reply:
I agree with @rhagarty I think it can be shortened to fewer steps.